crates/c-api/src/func.rs

```diff
@@ -217,8 +219,13 @@ fn _wasmtime_func_call(
     let result = panic::catch_unwind(AssertUnwindSafe(|| func.call(&params)));
     match result {
         Ok(Ok(out)) => {
-            for (slot, val) in results.iter_mut().zip(out.iter()) {
-                *slot = wasm_val_t::from_val(val);
+            for (slot, val) in results.iter_mut().zip(out.into_vec().into_iter()) {
+                unsafe {
```

Comment:
I think that writing to `&mut MaybeUninit<T>` is actually a safe operation, so could the unsafety be encapsulated in a helper method for now while we wait for `MaybeUninit::write`? That should help reduce the creep of `unsafe` into some of these functions.